src/test/kotlin/com/zepben/evolve/database/sqlite/metrics/MetricsDataSourceWriterTest.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2025 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.database.sqlite.metrics
+
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import com.zepben.evolve.database.sqlite.common.SchemaUtils
+import com.zepben.evolve.database.sqlite.common.TableVersion
+import com.zepben.evolve.database.sqlite.extensions.configureBatch
+import com.zepben.evolve.database.sqlite.metrics.tables.tableMetricsVersion
+import com.zepben.evolve.metrics.IngestionJob
+import com.zepben.testutils.exception.ExpectException.Companion.expect
+import io.mockk.*
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.slf4j.Logger
+import java.sql.Connection
+import java.util.*
+import javax.sql.DataSource
+
+internal class MetricsDataSourceWriterTest : MetricsSchemaTest() {
+
+    @AfterEach
+    internal fun unmockAll() {
+        clearAllMocks()
```

Comment:
todo: prefer block mocking rather than relying on `clearAllMocks` in a `@AfterEach` or `@AfterAll` as you don't have to go an look elsewhere to ensure things are being un-mocked correctly (less cognative load on reader of the code)